lib/templates/lab2/lab2.dart

```diff
@@ -0,0 +1,92 @@
+class Book {
+  String title;
+  String author;
+  int numberOfPages;
+
+  Book(this.title, this.author, this.numberOfPages);
+  void bookDetails() { 
+      // ignore: avoid_print
+      print("title: [$title], author: [$author], pages: [$numberOfPages]");
+    }
+}
+
+class FictionBook extends Book {
+  String genre;
+
+  FictionBook(super.title, super.author, super.numberOfPages, this.genre);
+  
+  @override
+  void bookDetails() {
+    // ignore: avoid_print
+    print("title: [$title], author: [$author], pages: [$numberOfPages], genre: [$genre]");
+  }
+}
+
+mixin JsonSerializable on Book {
+  // your code here
+  Map<String, dynamic> toJson() {
+    // your code here
+    return {"title": super.title,
+      "author": super.author,
+      "numberOfPages": super.numberOfPages};
+  }
+}
+
+class UpdatedBook extends Book with JsonSerializable {
```

Comment:
why we need one more class? It is not in the task, JsonSerializable should be mixin for Book